compiler/core/src/algorithms/bhp.rs

```diff
@@ -14,16 +14,16 @@
 // You should have received a copy of the GNU General Public License
 // along with the Leo library. If not, see <https://www.gnu.org/licenses/>.
 
-use crate::algorithms::{CoreFunction, ALL_TYPES};
+use crate::algorithms::CoreFunction;
 use leo_ast::Type;
 
 pub struct BHP256Hash;
 
 impl CoreFunction for BHP256Hash {
     const NUM_ARGS: usize = 1;
 
-    fn first_arg_types() -> &'static [Type] {
-        &ALL_TYPES
+    fn first_arg_is_allowed_type(type_: &Type) -> bool {
+        !matches!(type_, Type::Mapping(_) | Type::Tuple(_) | Type::Err | Type::Unit)
```

Comment:
Are mappings allowed as "regular" Leo values, that can be passed to hash and commit functions? I thought they were only "special" values that exist on chain?

Reply:
That's right! 
Mappings can only exist as top-level declarations in the Leo program.
I specified it in this manner to be complete w.r.t the internal compiler types.